### PR TITLE
Disable connection pooling for docker client.

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
By default, `hyper` uses connection pool that can grow till `usize::MAX`. This, we believe, could cause some issues when heavy [GetModuleLogs](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-retrieve-iot-edge-logs?view=iotedge-2020-11) are frequently executed.

Also, connection pool does not have benefits for local UDS connections anyways.

This probably should go in after 1.2.4 is released.

### CODE REVIEW

File: edgelet/edgelet-docker/src/runtime.rs [hunk 1]
```diff
@@ -1122,8 +1122,10 @@ impl Authenticator for DockerModuleRuntime {
 
 fn init_client(docker_url: &Url) -> Result<DockerClient<UrlConnector>> {
     // build the hyper client
-    let client =
-        Client::builder().build(UrlConnector::new(docker_url).context(ErrorKind::Initialization)?);
+    let client = Client::builder()
+        // we don't need connection pool'ing for local docker socket.
+        .max_idle_per_host(1)
```
Comment (arsing) [inline edgelet/edgelet-docker/src/runtime.rs:1127]: Does `0` work?

Comment (vadim-kovalyov) [reply edgelet/edgelet-docker/src/runtime.rs]: Honestly... don't know why I picket `1`:). Looking more at the hyper code, I realized that I need to set `keep_alive(false)` as well to completely disable connection pool.

File: edgelet/edgelet-docker/src/runtime.rs [hunk 2]
```diff
@@ -1122,8 +1122,11 @@ impl Authenticator for DockerModuleRuntime {
 
 fn init_client(docker_url: &Url) -> Result<DockerClient<UrlConnector>> {
     // build the hyper client
-    let client =
-        Client::builder().build(UrlConnector::new(docker_url).context(ErrorKind::Initialization)?);
+    let client = Client::builder()
+        // we don't need connection pool'ing for local docker socket.
+        .keep_alive(false)
```
Comment (varunpuranik) [inline edgelet/edgelet-docker/src/runtime.rs:1127]: Do you need to set keep_alive here as well, then? https://github.com/Azure/iot-identity-service/blob/main/http-common/src/connector.rs#L242

Comment (arsing) [reply edgelet/edgelet-docker/src/runtime.rs:1127]: There's no keep-alive in new hyper, so it's only a 1.2 problem.

Comment (vadim-kovalyov) [reply edgelet/edgelet-docker/src/runtime.rs:1127]: Correct ^

Comment (varunpuranik) [reply edgelet/edgelet-docker/src/runtime.rs:1127]: Got it.. thanks..

**Review by arsing — APPROVED** [review]

**Comment by vadim-kovalyov** [comment]
@varunpuranik is this good to go now? (1.2.4 is out...)

**Comment by varunpuranik** [comment]
> @varunpuranik is this good to go now? (1.2.4 is out...)

Yes it is..

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**